src/test/shell/integration/run_test.sh

```diff
@@ -353,5 +353,41 @@ EOF
   grep "Run Forest run" bazel-bin/a/output || fail "Output file wrong"
 }
 
+function test_sh_binary_data() {
+  cat > BUILD <<EOF
+load("//:whatever.bzl", "whatever")
+whatever()
+EOF
+
+  cat > whatever.bzl <<EOF
+def _catter_impl(ctx):
+    ctx.actions.write(ctx.outputs.out, "#!/bin/sh\ncat xyzzy.txt && echo Laputan Machine")
```

Comment:
This isn't right. xyzzy.txt is in the runfiles, but not guaranteed to be in the source tree. You need to use the runfiles library to find the right path.

Reply:
The "runfiles library" Ulf mentions is [here](https://github.com/bazelbuild/bazel/blob/cffe0e30e4949dbf2a68fb6f2747ffe14333c8a7/tools/bash/runfiles/runfiles.bash#L23).

Reply:
I tried doing this by copying the stanza linked by @laszlocsomor to the shell script generated through `_catter_impl()` and adding the dependency on `@bazel_tools//tools/bash/runfiles`. Unfortunately, this causes the test to fail with:

    ERROR: cannot find @bazel_tools//tools/bash/runfiles:runfiles.bash

I added `find .` to confirm that `runfiles.bash` is actually present, which it is. The problem seems to be that the `RUNFILES_*` environment variables don't get updated. The variables still contain the values set by the execution of `run_test.sh` by the copy of Bazel running the unit test.

Would there be a simpler way to test this reliably/correctly?

Reply:
Maybe bazel run isn't setting the RUNFILES_* env variables? Does it work with bazel test?

Reply:
No, bazel run does not set RUNFILES_*

Reply:
@EdSchouten , the stanza's purpose is to find the runfiles directory or manifest even if the RUNFILES_* envvars undefined. And apparently it doesn't. That's concerning. Let me try to repro this on my Linux box.

Reply:
Sorry, unfortunately I cannot look at this today.